subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/DefaultCurrentFileCollectionFingerprint.java

```diff
@@ -36,25 +39,45 @@
     private final String identifier;
     private final FileSystemSnapshot roots;
     private final ImmutableMultimap<String, HashCode> rootHashes;
+    private final HashCode strategyConfigurationHash;
     private HashCode hash;
 
     public static CurrentFileCollectionFingerprint from(FileSystemSnapshot roots, FingerprintingStrategy strategy) {
+        return from(roots, strategy, null);
+    }
+
+    public static CurrentFileCollectionFingerprint from(FileSystemSnapshot roots, FingerprintingStrategy strategy, @Nullable  FileCollectionFingerprint candidate) {
```

Comment:
No need for `@Nullable`.

Reply:
I don't think `FileCollectionFingerprint.EMPTY` is a better candidate than `null`, indicating that we don't have a previous candidate to re-use.

Reply:
You are passing `ImmutableMap.of()` in other places instead of `null`, and it works pretty well. I think for the test in question it also would work well.

Reply:
FTR, we agreed that `null` is the right solution here.